### PR TITLE
Issues/15

### PR DESCRIPTION
**Pull request by cameleeck**

### CODE REVIEW

**Comment by cameleeck** [comment]
Will squash after review.

File: accurest-core/src/main/groovy/io/coderate/accurest/TestGenerator.groovy [hunk 1]
```diff
@@ -42,83 +47,55 @@ class TestGenerator {
...
-
-		directory.listFiles().grep({ File file -> !file.isDirectory()}).each {
-			clazz.addMethod(createTestMethod(it, configProperties.targetFramework))
-		}
-		return clazz.build()
+	private boolean isJson(String name) {
+		"json".equalsIgnoreCase(NamesUtil.afterLastDot(name))
 	}
 
 	public static void main(String[] args) {
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/coderate/accurest/TestGenerator.groovy]: Exclude to test

File: accurest-core/src/main/groovy/io/coderate/accurest/config/AccurestConfigProperties.groovy [hunk 2]
```diff
@@ -10,6 +10,7 @@ class AccurestConfigProperties {
 	String basePackageForTests = 'io.codearte.accurest.tests'
 	String baseClassForTests
 	String ruleClassForTests
+    List<String> ignoredFiles = []
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/coderate/accurest/config/AccurestConfigProperties.groovy]: Format

File: accurest-core/src/test/groovy/io/codearte/accurest/TestGeneratorSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			TestGenerator testGenerator = new TestGenerator(properties, classGenerator, Stub(FileSaver))
+			classGenerator.buildClass(_, _, _) >> "sample"
+		when:
+			testGenerator.generateTestClasses(resource.parentFile, "com.ofg")
+		then:
+			1 * classGenerator.buildClass({ it.size() == 1 }, 'other', _) >> "sample.json"
+	}
+
+	@spock.lang.Ignore
+	def "Should create test for file in base directory"() {
```
Comment (jkubrynski) [inline accurest-core/src/test/groovy/io/codearte/accurest/TestGeneratorSpec.groovy]: You can remove it

File: accurest-core/src/main/groovy/io/coderate/accurest/TestGenerator.groovy [hunk 4]
```diff
@@ -1,40 +1,43 @@
...
-import static io.coderate.accurest.util.NamesUtil.afterLast
-import static io.coderate.accurest.util.NamesUtil.capitalize
-
 /**
  * @author Jakub Kubrynski
  */
 class TestGenerator {
 
-	private final String targetDirectory
+	public final NotFileFilter notWildcard
```
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/coderate/accurest/TestGenerator.groovy]: Why is it public?

File: build.gradle [hunk 5]
```diff
@@ -42,6 +46,13 @@ subprojects {
 }
 
 project(':accurest-core') {
+	dependencies {
+		compile 'commons-io:commons-io:2.4'
+		compile 'org.codehaus.plexus:plexus-utils:3.0.21'
+		testCompile 'cglib:cglib-nodep:2.2'
```
Comment (jkubrynski) [inline build.gradle:51]: Why do we need those dependecies?

Comment (cameleeck) [inline build.gradle:51]: Commons-io to wildcard matching files, plexus-utils for ant-style directory exclusions, and cglib/objenesis for spock mocking.

**Pull request closed without merging**